src/main/java/seedu/todo/logic/commands/TagCommand.java

```diff
@@ -120,6 +124,8 @@ public CommandResult execute() throws ValidationException {
         //Performs the actual execution with the data
         if (isAddTagsToTask()) {
             model.addTagsToTask(displayedIndex, tagsToAdd);
+            ImmutableTask task = model.getObservableList().get(displayedIndex - INDEX_OFFSET);
```

Comment:
Wrong model? Might only exist on the underlying layer. This model refers to the visible layer.